### PR TITLE
Refactoring School library

### PR DESCRIPTION
**Pull request by rachaouldbabaali**
in this pull request, we have done the following:

- [x] Ensure YAGNI, KISS, and DRY rules and Ruby style guides are respected.

- [x] Followed SOLID principles.
- [x] Paid special attention to the "Single-responsibility Principle".
- [x] Ensure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
- [x] Extracted some pieces of logic into a separate class(es).

### CODE REVIEW

File: classes/teacher.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'pry'
```
Comment (zieeco) [inline classes/teacher.rb]: - This line makes your app throw an error and prevents it to run. Please, can you fix this, otherwise, your app works fine.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @rachaouldbabaali,



Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️
- Console app works correctly :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions

- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or commthat the code review was not fairents in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rachaouldbabaali** [comment]
Thank you @zieeco, I truly appreciate your time and efforts in reviewing our code

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Team  

Your project is complete! Great job🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- works as expected
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rachaouldbabaali** [comment]
thank you @tufoinnkuo10, I truly appreciate your time and efforts in reviewing our code

**Pull request merged**